sympy/physics/wigner.py

```diff
@@ -216,8 +216,12 @@ def wigner_3j(j_1, j_2, j_3, m_1, m_2, m_3):
     - Jens Rasch (2009-03-24): initial version
     """
 
-    j_1, j_2, j_3, m_1, m_2, m_3 = map(_int_or_halfint,
-                                       [j_1, j_2, j_3, m_1, m_2, m_3])
+    j_1 = sympify(j_1)
```

Comment:
Does this still check whether the value is an integer or a half-integer?

Reply:
On manual testing it works for both integers and half integers.

Reply:
What if the value is not an integer or half integer? The purpose of the `_int_or...` function is to reject values that are not half integers.

Reply:
Oh, the expected behaviour in the documentation is for it to throw a ValueError, right now my code is returning 0 for these invalid cases. I need to further improve on this

Reply:
New commits have been added.

Reply:
All three cases - integers, half integers and neither ints or half ints are now handled, the function raises a ValueError as expected from the docs in the lattermost case. 3 tests are added to cover the error raising, the numpy float and half ints as well.